src/libbson/src/bson/bson-iter.c

```diff
@@ -1357,8 +1357,11 @@ bson_iter_dup_utf8 (const bson_iter_t *iter, /* IN */
    BSON_ASSERT (iter);
 
    if ((str = bson_iter_utf8 (iter, &local_length))) {
-      ret = bson_malloc0 (local_length + 1);
```

Comment:
Does the PoC in CDRIVER-5602 crash in `bson_iter_dup_utf8`? When running locally, I observe a crash on this line (for the same reason as CDRIVER-5601):

```c
BSON_APPEND_UTF8(b, "iamkey", payload);
```

If the PoC of CDRIVER-5602 does not crash in `bson_iter_dup_utf8`, I suggest asking for more detail in the ticket. If the crash is not reproducible, this fix may not be as well motivated.

Reply:
It does not crash in `bson_iter_dup_utf8`. Running the PoC in gdb against the base commit of this PR (a91e1893c685638b5192875d06bd95cf65f67d01) produces this backtrace at the point of the crash:

```
Program received signal SIGSEGV, Segmentation fault.
__memcpy_avx_unaligned_erms ()
    at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:841
841     ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S: No such file or directory.
(gdb) bt
#0  __memcpy_avx_unaligned_erms ()
    at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:841
#1  0x00007ffff7fa2243 in _bson_append_va (args=0x7fffffffd958,
    first_data=0x7ffef7d91010 's' <repeats 200 times>..., first_len=4294967294, n_pairs=1,
    n_bytes=1294903057, bson=0x555555559700)
    at /home/roberto/src/mongodb/mongo-c-driver.git/src/libbson/src/bson/bson.c:332
#2  _bson_append (bson=bson@entry=0x555555559700, n_pairs=n_pairs@entry=6,
    n_bytes=n_bytes@entry=11, first_len=first_len@entry=1,
    first_data=first_data@entry=0x7ffff7fb1833 <type> "\002")
    at /home/roberto/src/mongodb/mongo-c-driver.git/src/libbson/src/bson/bson.c:404
#3  0x00007ffff7fa3e33 in bson_append_utf8 (bson=0x555555559700,
    key=0x55555555606b "iamkey", key_length=<optimized out>,
    value=0x7ffef7d91010 's' <repeats 200 times>..., length=-2)
    at /home/roberto/src/mongodb/mongo-c-driver.git/src/libbson/src/bson/bson.c:1511
#4  0x000055555555530d in main ()
```

I'll request additional detail in the ticket.